### ISSUE TITLE
Backspace issue when running gdb from Windows console

### ISSUE THREAD

**Issue by Erbzur** [brechtsanders/winlibs_mingw#9]
**1.** Deleting gdb input command with the backspace key has a display problem, as the pic shown below:
![image](https://user-images.githubusercontent.com/39868528/79106085-645ac400-7da4-11ea-8753-aa7e041a2b2c.png)
The red mark is the actual cursor position. It seems that the backspace key makes the cursor run away and still keeps the gdb command text on console, although the command is actually deleted.

**2.** I was told that gcc 9.2+ had fixed <random_device> (in c++11 random lib) problem, but my test shows the problem still exists. 
```
random_device r;
cout << r.entropy() << endl;
```
Executing the code above will still print **zero** value on my machine, which means I can not use <random_device> to generate the real random number.

**Any idea?**

**Comment by brechtsanders** [comment]
**1.**
I built GDB against ncurses, and it worked fine under the MSYS2 shell.
But in my experience PDCurses works better on Windows, so I will try to build against PDCurses next time.

**2.**
Sounds like a question for the GCC team. I just compile their code for Windows without sourcecode changes...

**Comment by Erbzur** [comment]
Okay, thanks for your builds and prompt reply :D

**Comment by brechtsanders** [comment]
After looking into this, I was able to reproduce the issues with the examples within the readline library.
My build of readline was accidentally againt PDCurses while I wanted it to link against ncurses (which is much more recent). I was able to fix this, so in the next release this should be resolved.

**Comment by brechtsanders** [comment]
I just rebuilt the whole thing and I'm still seeing the backspace issue in gdb.
Next I will try to remove all traces of PDCurses and replace ncurses in all dependancies, but I will need some time for that.

**Comment by brechtsanders** [comment]
I just noticed that running the backspace problem in GDB is there when running in a windows console (like Command Prompt), but in a shell in mintty it works fine.

I was able to reproduce the problem when running `gdb` in a shell started with:
```
D:\msys64\usr\bin\bash.exe --noprofile --rcfile profile_settings.sh -i
```
while it works fine in a shell started in mintty like this
```
D:\msys64\usr\bin\mintty.exe --exec D:\msys64\usr\bin\bash.exe --noprofile --rcfile profile_settings.sh -i
```

**Comment by Erbzur** [comment]
Well, that's true. The "bash" uses a native cmd way to perform shell behaviors. I encountered another horrible problem about gdb when using the previous version of mingw64 builds. It's also due to cmd. I finally change the architecture from sjlj to seh to solve the problem.
Now we know that cmd is a pretty of shit, but if discarding cmd, the independent builds of mingw64 seems useless. Why not directly use msys2? (Personally I don't like it)
Anyway, my report is actually a tolerable problem. No need to waste too much time on it bro. :)

**Comment by brechtsanders** [comment]
I usually use MSYS2, so that's why I hadn't noticed the backspace issue in gdb.
Otherwise I use Code::Blocks to debug using gdb, so then I have no command line interaction with gdb.
Anyway, I just released a new build with gdb version 9.2 which was just released and the backspace issue is still there.
